runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java

```diff
@@ -79,7 +76,7 @@ public static FnApiControlClientPoolService offeringClientsToPool(
       // discarded, which should be performed by a call to #shutdownNow. The remote caller must be
       // able to handle an unexpectedly terminated connection.
       vendedClients.add(newClient);
```

Comment:
<!--thread_id:cc_182896326_t; commit:1fd121da1417624b3b84f0300648251da64b9cb5; resolved:0-->
<!--section:context-quote-->
> **angoenka** wrote:
> Incoming client connections are not serialized and we can get multiple client calls in parallel.

<!--section:body-->
Ack. I've added synchronization around `vendedClients` itself. It may be worth synchronizing the entire close method now that we're synchronizing over the bulk of its work. @tgroh What do you think?